remove "$" from scalac generated anonymous term

godenji opened the pull request:
In the event that the user elects *not* to supply table aliases (e.g. `User join UserRole on(_.id == _.userId)` scalac generates anonymous terms in the form of `x$`, which `freshIdent` then generates unique aliases for (in this case, `x$1` and `x$2`).

Some Quill users may prefer the more concise `on` syntax, but still want generated sql to be readable. Removing "$" from `Ident` term name provides similar readability to manually supplied table aliases.

fwbrasil commented on quill-core/src/main/scala/io/getquill/norm/capture/AvoidAliasConflict.scala:
```diff
@@ -40,11 +40,14 @@ private case class AvoidAliasConflict(state: Set[Ident])
-  private def freshIdent(x: Ident): Ident =
+  private def freshIdent(x: Ident): Ident = {
+    // strip "$" from scalac generated anonymous term like `x$`
+    val clean = x.copy(name = x.name.replace("$", ""))
```
Could you extract a method to avoid the comment?

fwbrasil commented:
Sounds good. Please also add a test to AvoidAliasConflictSpec

godenji commented:
Hmmm, failing on [`QueryMacroSpec`](https://github.com/godenji/quill/blob/master/quill-core/src/test/scala/io/getquill/source/QueryMacroSpec.scala#L10) "runs non-binded action" test. Investigating...

fwbrasil commented:
It's because the match value isn't being normalized and thus doesn't have clean identifiers. Maybe it'd be better to implement this change in Parser.scala, so the AST is produced already clean.

godenji commented:
Yes, the $ stripping has to occur at an earlier point.

godenji commented:
@fwbrasil done, moved Ident name cleaner to Parsing.scala. Added a test for anonymous properties in QuotationSpec and updated FlattenOptionOperationSpec accordingly.

fwbrasil commented:
You're on a roll! :)

godenji commented:
Compared to that other query DSL in the Scala ecosystem, Quill is *much* easier to work on ;-)

BTW, have a WIP branch where I'm exploring SQL-like joins. The basic idea is that instead of:
```
val q = quote{
  for{
    (((u,ur), r), c) <- User join UserRole on (_.id == _.userId) join Role on ((x,r)=>
      x._2.roleId == r.id
    ) join Cart on((x, c)=> x._1._1.id == c.userId)
  } yield(u,ur,r,c)
}
```
we would have something like:
```
val q = quote{
  for{
    (u,ur) <- User join UserRole on (_.id == _.userId) 
    r <- Role join(_.id == ur.roleId)
    c <- Cart join(_.userId == u.id)
  } yield(u,ur,r,c)
}
```
In the former example it's difficult to work out what's happening due to the deeply nested tuples, and that's just a simple 4 table join, real world join queries will quickly turn into nested tuple soup.

In order for the "ideal" version to work we'll need to think about the requirement that a join always be a pair of tables. With plain sql the first set of tables in a join table are a pair, but thereafter it's single tables joined to previous table(s) via the `on` clause.

Anyway, this will be in a separate PR if I manage to get it working...
